tests/test_llmrails.py

```
@@ -1475,3 +1477,71 @@ def test_embedding_model_no_backfill_when_no_embeddings_model():
+            pytest.param({"rails": ["input", "output"]}, False, USER_ASSISTANT_MSG, 0, id="input-output-no-warn"),
+            pytest.param(None, True, USER_MSG, 0, id="llm-no-options-no-warn"),
+            pytest.param({"rails": ["input", "dialog"]}, True, USER_MSG, 0, id="llm-dialog-enabled-no-warn"),
+        ],
+    )
+    @pytest.mark.asyncio
+    async def test_warning_behavior(self, no_main_llm_config, caplog, options, has_llm, messages, expected_warnings):
+        llm = FakeLLMModel(responses=["Hello!"]) if has_llm else None
+        rails = LLMRails(no_main_llm_config, llm=llm)
+        with caplog.at_level(logging.WARNING, logger="nemoguardrails.rails.llm.llmrails"):
+            if expected_warnings > 0:
+                with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)